client/components/place/placesOnMap.factory.js

```diff
@@ -94,16 +109,16 @@ angular.module('mapModule')
         coords[1] = place.location.coordinates[0];
         coordsArray[index] = coords;
       });
-      trackForAdding = polyline(coordsArray, color).addTo(map);
+      trackForAdding = polyline(coordsArray, color).addTo(placesOnMap.map);
       tracks.push([trackForAdding, track.type]);
     };
 
     var removeTrack = function(track) {
       if (this == 'all') {
```

Comment:
:eyes: You can just move `forEach` here and pass 'types' as argument. 
This will simplify this function, because `this` is very implicit value here.